### PR TITLE
Close #21810

### LINKED ISSUE #21810: transformer

### PR DESCRIPTION
**Pull request by DevBhuyan**
Closes #21810 
https://github.com/unifyai/ivy/issues/21810

From ToDo list: #14945 
https://github.com/unifyai/ivy/issues/14945 

Update: 
1. Added transformer layer in ~/ivy/ivy/stateful/layers.py
2. Added test `test_transformer_layer` in ~/ivy/ivy_tests/test_ivy/test_stateful/test_layers.py
3. Added custom composite strategy to generate data (ref. transformer_data())

This is my first PR. I tried to make it as appropriate as possible. Please let me know if there are any modifications that you'd suggest. Thank you!😊

### CODE REVIEW

**Comment by yopknopixx** [comment]
Hey, @DevBhuyan could you please link the open issue in the PR description?

**Comment by DevBhuyan** [comment]
Sure @yopknopix. I have updated the initial comment accordingly. I've included the open issue #21810 as well as the corresponding ToDo list. Thank you!

**Comment by yopknopixx** [comment]
Can you pull the latest changes from master? There seems to be some issue with the testing pipeline, no tests are being run. Thanks!

**Comment by DevBhuyan** [comment]
Sure, I've synced my version of the fork again. I can see that one of the checks `run_tests (1) (pull_request) ` has failed. This has been constantly failing for all of the previous commits. I guess it's safe to ignore that one, as mentioned here:

> Thanks for contributing to Ivy! 😊👏 Here are some of the important points from our Contributing Guidelines 📝: 1. Feel free to ignore the `run_tests (1)`, `run_tests (2)`, … jobs, and only look at the `display_test_results` job. 👀........ 

I hope it helped. Please let me know if there is still an issue. Thank you 😇

File: ivy/stateful/layers.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 from ivy.func_wrapper import handle_nestable
 from ivy.stateful.initializers import GlorotUniform, Zeros
 from ivy.stateful.module import Module
+from ivy.stateful.norms import LayerNorm
+import numpy as np
```
Comment (yopknopixx) [inline ivy/stateful/layers.py:9]: @DevBhuyan You should not use NumPy for this. Please use ivy functions.

**Comment by yopknopixx** [comment]
@DevBhuyan it seems for some reason the tests are not running at all. As for `run_tests (1) (pull_request) ` you're correct, but that's only when tests are running and failing.

**Comment by DevBhuyan** [comment]
Oh I see, please let me know if there's something I need to do to help resolve the testing issue. Meanwhile I'll fix the code and commit again. Thank you 😇

**Comment by yopknopixx** [comment]
Can you try creating a new branch on your fork from `unifyai:main`, implementing these changes, and creating a new PR?

**Comment by DevBhuyan** [comment]
Sure! I'll do that then.

**Comment by yopknopixx** [comment]
Hey @DevBhuyan, we can close this PR. The tests seem to be working fine on the new one.

**Comment by DevBhuyan** [comment]
Alright, thanks 😇. But I guess the new PR is being handled by someone else, is it?

**Pull request closed without merging**